### ISSUE TITLE
[Suggestion] Call Block#getActualState for determination of applicable BlockRenderLayers

### ISSUE THREAD

**Issue by XFactHD** [MinecraftForge/MinecraftForge#4421]
While developing a block that uses a Forge multilayer model in a specific state and a normal vanilla model in the other states, I noticed that it is not possible to determine if a certain BlockRenderLayer is applicable for this IBlockState in `Block#canRenderInPass(IBlockState)` when the decision is based on a value that is stored in the TileEntity. When I looked at `RenderChunk#rebuildChunk(float x, float y, float z, ChunkCompileTaskGenerator generator)` I noticed that `Block#getActualState(IBlockState state, IBlockAccess worldIn, BlockPos pos)` is not taken into consideration for this.
As it is impossible to call `Block#getActualState(IBlockState state, IBlockAccess worldIn, BlockPos pos)` in `Block#canRenderInPass(IBlockState)` for obvious reasons, I would suggest that RenderChunk calls `Block#getActualState(IBlockState state, IBlockAccess worldIn, BlockPos pos)` on the IBlockState used to determine the applicable BlockRenderLayers.
The block where I noticed this can be found [here](https://github.com/XFactHD/RainbowSixSiegeMC/blob/master/1.10.2/src/main/java/XFactHD/rssmc/common/blocks/gadget/BlockBlackMirror.java). While this is a 1.10.2 mod, I have confirmed that the behaviour is theoretically the same in 1.12.2 (Forge 1.12.2-14.23.0.2491).

**Comment by grondag** [comment]
I have a similar use case and would use this if implemented.  To workaround I implemented different block states for different render combinations. This consumes a few more block IDs but I did not want to make a core mod. Another alternative would be to always return true but then RenderChunk wastes cycles asking my block model for empty quad lists.  My code is [here](https://github.com/grondag/Hard-Science/blob/master/src/main/java/grondag/hard_science/superblock/block/SuperBlock.java).
